### PR TITLE
add paginator component and updateQueryParams helper

### PR DESCRIPTION
**Pull request by mcbouslog**
paginator component based on [Talk paginator](https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/talk/lib/paginator.cjsx)
updateQueryParams helper based on [Talk updateQueryParams](https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/talk/lib/update-query-params.coffee)

Questions
~~A. compared to PFE version, this does not include `if page is pageCount then pageCount else page + 1` and similar for previous/next default functions because buttons are disabled if applicable~~
B. please review arrow(s) styling, a little hacky to position vertically, might be a better way
~~C. `itemCount` and `totalItems` duplicative?~~
D. three possible scenarios on how to deal with changes:
   1. router provided
   2. onChange provided
   3. onClickNext, onClickPrev provided (firstAndLast=false)

if combination of 1-3 provided then there'd be conflicts or something provided would be ignored, should that be messaged via console warnings or something similar?

please squash and merge

### CODE REVIEW

**Comment by mcbouslog** [comment]
I pushed up a `paginator-testing` branch in the `pfe-lab` repo with a container utilizing the paginator to potentially aid in testing if interested: https://github.com/zooniverse/pfe-lab/blob/paginator-testing/src/modules/organizations/containers/projects-container.jsx

File: src/components/paginator.jsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  firstAndLast: true,
+  firstLabel: <span><span className="paginator-icon">&laquo;</span> FIRST</span>,
+  itemCount: false,
+  lastLabel: <span>LAST <span className="paginator-icon">&raquo;</span></span>,
+  nextLabel: <span>NEXT <span className="paginator-icon">&rsaquo;</span></span>,
+  page: 1,
+  pageKey: 'page',
+  pageSelector: true,
+  previousLabel: <span><span className="paginator-icon">&lsaquo;</span> PREVIOUS</span>,
+  totalItems: undefined,
```
Comment (srallen) [inline src/components/paginator.jsx]: Let's just not set a default prop for this. Setting as undefined is confusing.

**Comment by mcbouslog** [comment]
updated related `pfe-lab` branch `paginator-testing`'s ZRC dependency to ZRC `paginator` branch.

**Comment by mcbouslog** [comment]
I think initial comments addressed, no urgency, but let me know if I missed anything or after had a chance for additional review.

**Comment by mcbouslog** [comment]
that's weird, deleted the lib folder, sounds good rgd plan on generating on version bump.

**Comment by srallen** [comment]
@eatyourgreens made a good suggestion on slack:
```Can we make it a rule that package lock changes get their own commit. Just to make them easier to delete or revert if necessary.```

File: src/components/paginator.jsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        </button>)}
+
+    </div>
+  );
+};
+
+Paginator.defaultProps = {
+  className: '',
+  firstAndLast: true,
+  firstLabel: <span><span className="paginator-icon">&laquo;</span> FIRST</span>,
```
Comment (srallen) [inline src/components/paginator.jsx]: Let's not hardcode these with uppercase. Maybe user lowercase and add a css style for these to use `text-transform: capitalize` which can be overwritten by css if needed.

File: src/css/paginator.styl [hunk 3]
```diff
@@ -0,0 +1,25 @@
+.paginator
+  font-size: .86em
+  line-height: 1.37em
+  text-align: center
+
+  .paginator-button
+    @extends $reset-button
+    margin: 5px
+
+    &:hover
```
Comment (srallen) [inline src/css/paginator.styl]: Add also `&:focus`. Generally hover and focus styles usually should be the same and if one is defined, the other also should be for accessibility and consistency.

**Pull request merged**